Maybe/MaybeEnumerableExtensions.cs

```diff
@@ -131,7 +131,7 @@ public static Maybe<T> MaybeGet<T>(this IList<T> source, int position)
 
                     if (enumerator.MoveNext())
                     {
-                        throw new InvalidOperationException("Sequence contains more than one element");
+                        throw new InvalidOperationException($"Maybe<IEnumerable<{typeof(T).Name}>>: Sequence contains more than one element");
```

Comment:
Changing the Exception message potentially means breaking changes in case someone is doing a try-catch specifically on the message text. So this will require a major release.

Reply:
I said major release but 1.1 probably works here. I meant to say "not a fix release".